### DISCUSSION TITLE
Reset specific field from array using useFieldArray

### DISCUSSION THREAD

**Discussion by FilipBrannlund** [react-hook-form/react-hook-form#6204] [Q&A] [answered]
Having following defaultValues:
```
    defaultValues: {
      people: [
        {
          id: 0,
          firstName: "Tom"
        },
        {
          id: 1,
          firstName: "John"
        },
        {
          id: 2,
          firstName: "Todd"
        }
      ]
    }
```

I want to reset ONE specific field in the array using `reset`
`reset()` resets all field

```
{fields.map((field) => (
        <div>
          <input key={field.id} {...register(`people.${field.id}.firstName`)} />
          <button onClick={() => resetField(field.id)}>
            Reset to default value
          </button>
        </div>
      ))}
```
Followed by the id that is passed in field with matching id should reset to default value. The other fields should not reset
```
  const resetField = (id) => {
    console.log(`Resetting field ${id} to it's default value...`);
    reset(); // Resets ALL fields, but I want to reset field with ${id}
  };
```

https://codesandbox.io/s/jolly-leaf-kok7t?file=/src/App.js

**Comment by bluebill1049** [comment, marked as answer]
use `setValue`

**Reply by FilipBrannlund** [reply]
Ok thanks. Could you check my CSB out. I'm having some issues that dirtyFields is not clearing after reseting value using setValue.
Scenario 1: (OK)
Type 200 - Shows that field was dirty
Change back to 100 - Field is not dirty

Scenation 2: (NOT OK)
Type 200 - Shows that field was dirty
Click reset btn. Fields get reseted to 100 but still shows as field is dirty

https://codesandbox.io/s/eloquent-haze-35wyt?file=/src/App.tsx

**Reply by bluebill1049** [reply]
> You can also set the shouldDirty parameter to true in order to set the field to dirty.

https://react-hook-form.com/api/useform/setvalue

**Reply by FilipBrannlund** [reply]
`shouldDirty = true ` solved the problem.
